### ISSUE TITLE
[FEATURE] Add Support for Regression Models (Uncertainty & Error Analysis)

### ISSUE THREAD

**Issue by Khanz9664** [Khanz9664/TrustLens#82]
#### **Problem**
TrustLens is currently optimized for classification tasks (providing Brier Scores, ECE, and Bias gaps). However, a significant portion of production ML involves **Regression** (predicting continuous values). Trusting a regression model requires more than just looking at $R^2$; it requires understanding **Uncertainty Calibration** and **Heteroscedasticity** (error variance).

#### **The Vision**
We want to enable `trustlens.analyze(model, X, y_true)` to automatically detect when a target is continuous and switch to the **Regression Pipeline**.

#### **Your Task**
Implement the following core metrics and visualizations in a new `trustlens/metrics/regression.py` module:

1.  **Expected Prediction Error (EPE) Distribution**:
    *   Instead of misclassification, report the distribution of absolute errors.
    *   **Metric**: Median Absolute Error (MedAE) and 90th percentile error.
2.  **Uncertainty Calibration (for Probabilistic Regression)**:
    *   If the model provides prediction intervals (e.g., Quantile Regression), implement **PICP (Prediction Interval Coverage Probability)**.
    *   A trustworthy model should have a coverage probability that matches the confidence level (e.g., 95% of points fall in the 95% interval).
3.  **Error-Confidence Correlation**:
    *   Measure the correlation between predicted variance (if available) and actual error.
4.  **Visualization**:
    *   `plot_residual_analysis()`: A plot showing residuals vs. predicted values to detect patterns (heteroscedasticity).
    *   `plot_error_distribution()`: A histogram of errors compared to a normal distribution.

#### **Expected API**
```python
import trustlens

# TrustLens should auto-detect regression based on y_true.dtype
report = trustlens.analyze(reg_model, X_val, y_val)

report.show()
# Outputs:
# Regression Analysis
# - Median Absolute Error: 1.24
# - Max Error: 15.3 (Potential Outlier Risk)
# - Uncertainty Coverage: 92% (Target: 95%) -> Warning: Under-confident
```

#### **Definition of Done**
- [ ] New `metrics/regression.py` with MedAE and Coverage metrics.
- [ ] New `visualization/regression_plots.py` with residual and error distribution plots.
- [ ] Logic in `api.py` to auto-dispatch to the regression pipeline if `y_true` is continuous.
- [ ] At least 3 unit tests using `sklearn.datasets.make_regression`.

---

## Completed
- [x] Regression metrics module
- [x] Error Distribution
- [x] PICP
- [x] Error-Variance Correlation

## Remaining
- [ ] Regression pipeline auto-dispatch
- [ ] TrustReport integration
- [ ] Residual analysis visualization
- [ ] Error distribution visualization
- [ ] Regression report rendering
- [ ] Regression trust score

**Comment by Whatsonyourmind** [comment]
Opened #141 implementing the metrics module — `error_distribution` (MedAE / p90 / max / MAE / RMSE + histogram), `prediction_interval_coverage` (PICP vs nominal, skips gracefully when no intervals are supplied since sklearn regressors expose only point predictions), and `error_variance_correlation` (Pearson/Spearman of predicted uncertainty vs actual |error|). 16 tests incl. `make_regression`; ruff/format/mypy clean; matches the existing metric conventions (raw arrays in, dict out, `status="skipped"` degradation).

Scoped to the metrics module — I deliberately left the `analyze()` auto-dispatch and the `prediction_contract`/backend changes (they also touch `report.py` and `trust_score.py`) plus the visualization functions as follow-ups, per the one-PR-per-feature guideline. Happy to take those next if this scoping works for you.

**Comment by Khanz9664** [comment]
@Whatsonyourmind Thanks for taking this on and for keeping the scope focused.

Regression support is one of the larger roadmap items for TrustLens, so I appreciate the decision to split the work into smaller architectural pieces rather than trying to land everything in a single PR.

The metrics layer is a sensible first step. I'll review the implementation in #141 on its own merits, and we can evaluate follow-up work (visualizations, report integration, pipeline dispatch, etc.) separately once the foundational metrics are in place.

Thanks for documenting the scope clearly and for keeping the implementation aligned with the existing metrics conventions.
